src/angular/form-elements/dropdown/dropdown.component.ts

```diff
@@ -1,28 +1,224 @@
+     */
     @Input() placeHolder:string;
+
+    /**
+     * Drop-down required flag
+     */
     @Input() required:boolean;
 
+
+    @Input() validate:boolean;
+
+    @Input() headless = false;
+
+    /**
+     * Selected option index
+     */
+    public selectedIndex: number;
+
+    /**
+     * Selected option label
+     */
+    public selectedLabel:string;
+
+    /**
+     * The selected drop-down group
+     */
+    public selectedDropDownGroup: IDropDownGroup;
+
+    /**
+     * Drop-down show/hide flag. default is false (closed)
+     * @type {boolean}
+     */
+    public show = false;
+
+    /**
+     * Error flag
+     * @type {boolean}
+     */
+    public error: boolean;
+
+    ngOnChanges(changes: SimpleChanges): void {
+        if(changes.value){
+            this.setValueIfExist(this.value);
+        }
+    }
+
+    public isValid(){
+        return !this.error;
+    }
+
+    /**
+     * Validate when required is enabled
+     */
+    public validateDropDown(): void{
+        if(!this.disabled && this.required && (!this.value || this.value === '')){
+            this.error = true;
+            return;
+        }
+        this.error = false;
+    }
+
+    /**
+     * Set option as selected and saves it's index on the list
+     * @param index - number
+     * @param option - IDropDownItem or string
+     */
+    public selectOption(index: number, option:any):void{
+        this.selectedIndex = index;
+        this.updateSelected(option.label || option, option.value || option);
+    }
+
+    /**
+     * Set drop-down group item as selected. Save the selected group as well as the value
+     * @param event
+     */
+    public selectGroupOption(event: IDropDownGroupResult){
+        const item = event.group.items.find((i) => event.value === i.value);
+        this.selectedDropDownGroup = event.group;
+        this.updateSelected(item.label, item.value);
+    }
+
+    /**
+     * Update the value and label of the drop down with new ones
+     * @param label
+     * @param value
+     */
+    private updateSelected(label: string, value: string, index?: number){
+        this.selectedLabel = label;
+        this.value = value;
+        this.baseEmitter.next(this.value);
+        this.selectedIndex = index || this.selectedIndex;
+        this.validateDropDown();
+    }
+
+    /**
+     * Get the value of selected drop-down group item
+     * @param dropdownGroup
+     * @returns {string}
+     */
+    public getDropDownGroupSelectedValue(dropdownGroup: IDropDownGroup):string{
+        if(this.selectedDropDownGroup && this.selectedDropDownGroup.title === dropdownGroup.title){
+            return this.value;
+        }
+        return null;
+    }
+
+    /**
+     * Check if our options are IDropDownGroup items
+     */
+    public isDropDownGroupList(): boolean{
+        return this.options && this.options.length && this.options[0].hasOwnProperty('title') || false;
```

Comment:
Again, you can't tell from the first item in the list something on the entire list.
Can't I combine both groups and items in the options?
Maybe the group is not a good approach for definition - you can define it as an option header, which affects design only, so there's no group of options at all. In the same manner you can define an option line, which will be designed as an horizontal line in the options list. But the options are never nested, and all of them have the same logic.